build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts

```diff
@@ -38,33 +38,18 @@ plugins {
 version =
     providers.fileContents(rootProject.layout.projectDirectory.versionTxt()).asText.get().trim()
 
-// Fail the build if Gradle is started with an old version of Java
-val MIN_JAVA_VERSION = JavaVersion.VERSION_21
-val CUR_JAVA_VERSION = JavaVersion.current()
-
-if (CUR_JAVA_VERSION.ordinal < MIN_JAVA_VERSION.ordinal) {
-    throw StopExecutionException(
-        "ERROR: Gradle is started with Java " +
-            CUR_JAVA_VERSION +
-            ". This project requires running Gradle with Java " +
-            MIN_JAVA_VERSION +
-            " or above." +
-            " Please check your JAVA_HOME and/or PATH and configure the default JDK to use Java version " +
-            MIN_JAVA_VERSION +
-            " or above."
+// Fail the build if Gradle is started with a Java version that does not exactly match
+val javaVersion = "21.0.1"
+val currentJavaVersion = providers.systemProperty("java.version").get()
+
+if (currentJavaVersion != javaVersion) {
+    throw RuntimeException(
+        "Gradle runs with Java $currentJavaVersion. This project requires Gradle to run with Java $javaVersion. " +
+            "\n - From commandline: change JAVA_HOME and/or PATH to point at Java $javaVersion installation." +
+            "\n - From IntelliJ: change 'Gradle JVM' in 'Gradle Settings' to point at Java $javaVersion installation."
```

Comment:
Same check as before, but for the **exact** version a a bit of rewording.

Reply:
Personally, I find it limiting to fix a particular version of JDK for the build as this creates an extra friction when trying to experiment with newer JDKs, or may create real problems when a JDK requires an urgent update (e.g. due to a security vulnerability or similar.) So I cannot say I like this fix.

However, I do understand the intention of this change. And personally, I'd prefer if a minor (or even major) version mismatch was treated as a warning rather than an error (with a proper minimum version check, of course - that should still result in an error.)

But as far as the original idea is concerned, from functional perspective the fix looks correct to me. So I'm approving it.